File: src/services/tree-sitter/__tests__/fixtures/swift/sample.swift
import Foundation

/// SwiftClass is a sample class.
class SwiftClass {
    var name: String

    init(name: String) {
        self.name = name
    }

    /// getName returns the name.
    func getName() -> String {
        return self.name
    }

    func calculate(a: Int, b: Int) -> Int {
        return a + b
    }
}

/// topLevelFunc is a top-level function.
func topLevelFunc(x: Int) -> Int {
    let obj = SwiftClass(name: "example")
    print(obj.getName())
    return obj.calculate(a: x, b: 10)
}

topLevelFunc(x: 5)

struct MyStruct {
    func structMethod() {
        print("struct")
    }
}

extension SwiftClass {
    func extensionMethod() {
        print("extension")
    }
}

File: src/services/tree-sitter/__tests__/fixtures/ruby/sample.rb
# RubyClass is a sample class.
class RubyClass
  attr_accessor :name

  def initialize(name)
    @name = name
  end

  # get_name returns the name.
  def get_name
    @name
  end

  def calculate(a, b)
    a + b
  end
end

# top_level_func is a top-level function.
def top_level_func(x)
  obj = RubyClass.new("example")
  puts obj.get_name
  obj.calculate(x, 10)
end

top_level_func(5)

module SampleModule
  def self.module_func
    "hello"
  end

  class NestedClass
    def nested_method
      "nested"
    end
  end
end
